From fe4e555ecfb0cee389c2b246c02229dd59ec3de4 Mon Sep 17 00:00:00 2001
From: Christian Walther <walther@indel.ch>
Date: Thu, 20 Feb 2014 17:13:02 +0100
Subject: [PATCH] Expose ScintillaWX methods DoDragEnter and DoDragLeave from
 wxStyledTextCtrl in addition to DoDragOver. These are
 needed to properly implement alternative wxDropTargets.

DoDragEnter does nothing particularly important, but omitting DoDragLeave leaves behind a drag insertion point indicator that hides the normal text insertion cursor.
---
 include/wx/stc/stc.h       |    6 ++++++
 interface/wx/stc/stc.h     |   10 ++++++++++
 src/stc/stc.cpp            |    9 ++++++++-
 src/stc/stc.cpp.in         |    9 ++++++++-
 src/stc/stc.h.in           |    6 ++++++
 src/stc/stc.interface.h.in |   10 ++++++++++
 6 files changed, 48 insertions(+), 2 deletions(-)

diff --git a/include/wx/stc/stc.h b/include/wx/stc/stc.h
index 7d7b11a..3209a69 100644
--- a/include/wx/stc/stc.h
+++ b/include/wx/stc/stc.h
@@ -4727,9 +4727,15 @@ public:
 #endif // !wxUSE_TEXTCTRL
 
 #ifdef STC_USE_DND
+    // Allow for simulating a DnD DragEnter
+    wxDragResult DoDragEnter(wxCoord x, wxCoord y, wxDragResult def);
+
     // Allow for simulating a DnD DragOver
     wxDragResult DoDragOver(wxCoord x, wxCoord y, wxDragResult def);
 
+    // Allow for simulating a DnD DragLeave
+    void DoDragLeave();
+
     // Allow for simulating a DnD DropText
     bool DoDropText(long x, long y, const wxString& data);
 #endif
diff --git a/interface/wx/stc/stc.h b/interface/wx/stc/stc.h
index b012089..0a02e21 100644
--- a/interface/wx/stc/stc.h
+++ b/interface/wx/stc/stc.h
@@ -6001,11 +6001,21 @@ public:
     bool LoadFile(const wxString& filename);
 
     /**
+       Allow for simulating a DnD DragEnter
+    */
+    wxDragResult DoDragEnter(wxCoord x, wxCoord y, wxDragResult defaultRes);
+
+    /**
        Allow for simulating a DnD DragOver
     */
     wxDragResult DoDragOver(wxCoord x, wxCoord y, wxDragResult defaultRes);
 
     /**
+       Allow for simulating a DnD DragLeave
+    */
+    void DoDragLeave();
+
+    /**
        Allow for simulating a DnD DropText
     */
     bool DoDropText(long x, long y, const wxString& data);
diff --git a/src/stc/stc.cpp b/src/stc/stc.cpp
index b51afa4..2d82a2c 100644
--- a/src/stc/stc.cpp
+++ b/src/stc/stc.cpp
@@ -4738,10 +4738,17 @@ bool wxStyledTextCtrl::LoadFile(const wxString& filename)
 #endif // !wxUSE_TEXTCTRL
 
 #if wxUSE_DRAG_AND_DROP
+wxDragResult wxStyledTextCtrl::DoDragEnter(wxCoord x, wxCoord y, wxDragResult def) {
+    return m_swx->DoDragEnter(x, y, def);
+}
+
 wxDragResult wxStyledTextCtrl::DoDragOver(wxCoord x, wxCoord y, wxDragResult def) {
-        return m_swx->DoDragOver(x, y, def);
+    return m_swx->DoDragOver(x, y, def);
 }
 
+void wxStyledTextCtrl::DoDragLeave() {
+    m_swx->DoDragLeave();
+}
 
 bool wxStyledTextCtrl::DoDropText(long x, long y, const wxString& data) {
     return m_swx->DoDropText(x, y, data);
diff --git a/src/stc/stc.cpp.in b/src/stc/stc.cpp.in
index 48e3ac8..900a2bc 100644
--- a/src/stc/stc.cpp.in
+++ b/src/stc/stc.cpp.in
@@ -626,10 +626,17 @@ bool wxStyledTextCtrl::LoadFile(const wxString& filename)
 #endif // !wxUSE_TEXTCTRL
 
 #if wxUSE_DRAG_AND_DROP
+wxDragResult wxStyledTextCtrl::DoDragEnter(wxCoord x, wxCoord y, wxDragResult def) {
+    return m_swx->DoDragEnter(x, y, def);
+}
+
 wxDragResult wxStyledTextCtrl::DoDragOver(wxCoord x, wxCoord y, wxDragResult def) {
-        return m_swx->DoDragOver(x, y, def);
+    return m_swx->DoDragOver(x, y, def);
 }
 
+void wxStyledTextCtrl::DoDragLeave() {
+    m_swx->DoDragLeave();
+}
 
 bool wxStyledTextCtrl::DoDropText(long x, long y, const wxString& data) {
     return m_swx->DoDropText(x, y, data);
diff --git a/src/stc/stc.h.in b/src/stc/stc.h.in
index 63fb35a..b8279aa 100644
--- a/src/stc/stc.h.in
+++ b/src/stc/stc.h.in
@@ -221,9 +221,15 @@ public:
 #endif // !wxUSE_TEXTCTRL
 
 #ifdef STC_USE_DND
+    // Allow for simulating a DnD DragEnter
+    wxDragResult DoDragEnter(wxCoord x, wxCoord y, wxDragResult def);
+
     // Allow for simulating a DnD DragOver
     wxDragResult DoDragOver(wxCoord x, wxCoord y, wxDragResult def);
 
+    // Allow for simulating a DnD DragLeave
+    void DoDragLeave();
+
     // Allow for simulating a DnD DropText
     bool DoDropText(long x, long y, const wxString& data);
 #endif
diff --git a/src/stc/stc.interface.h.in b/src/stc/stc.interface.h.in
index 0aae470..220f90b 100644
--- a/src/stc/stc.interface.h.in
+++ b/src/stc/stc.interface.h.in
@@ -279,11 +279,21 @@ public:
     bool LoadFile(const wxString& filename);
 
     /**
+       Allow for simulating a DnD DragEnter
+    */
+    wxDragResult DoDragEnter(wxCoord x, wxCoord y, wxDragResult defaultRes);
+
+    /**
        Allow for simulating a DnD DragOver
     */
     wxDragResult DoDragOver(wxCoord x, wxCoord y, wxDragResult defaultRes);
 
     /**
+       Allow for simulating a DnD DragLeave
+    */
+    void DoDragLeave();
+
+    /**
        Allow for simulating a DnD DropText
     */
     bool DoDropText(long x, long y, const wxString& data);
-- 
1.7.5.4

